source/parser/x86_64/register_class.hpp

```diff
@@ -0,0 +1,21 @@
+// Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+// Spack Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+namespace smeagle::x86_64 {
+
+  // A register class for AMD64 is defined on page 16 of the System V abi pdf
+  enum RegisterClass {
```

Comment:
Since this possibly has larger scope now, let's change it from `enum` to `enum class` to prevent the enumerated names from clogging up the namespace.

Reply:
done.